crates/dojo-core/src/world.cairo

```diff
@@ -595,9 +621,9 @@ mod world {
                 if i >= new_state.len() {
                     break;
                 }
-                let base = starknet::storage_base_address_from_felt252(*new_state[i]);
+                let base = starknet::storage_base_address_from_felt252(*new_state.at(i).key);
                 starknet::storage_write_syscall(
-                    0, starknet::storage_address_from_base(base), *new_state[i + 1]
+                    0, starknet::storage_address_from_base(base), *new_state.at(i + 1).value
```

Comment:
Here it's only `i` if we use the struct and not `i + 1`.

Reply:
fixed, I've missed that